src/bevent.c

```diff
@@ -506,7 +506,9 @@ int odict_encode_bevent(struct odict *od, struct bevent *event)
 		if (hdr)
 			err = odict_pl_add(od, "contact", &hdr->val);
 
-		err |= odict_pl_add(od, "display", &msg->from.dname);
+		if (pl_isset(&msg->from.dname))
+			err = odict_pl_add(od, "display", &msg->from.dname);
+
```

Comment:
this will overwrite `err` from previous line where it is set.

the correct thing to do is to check `err` or to use `err |=`